src/SixLabors.Fonts/Tables/AdvancedTypographic/GPos/LookupType5SubTable.cs

```diff
@@ -0,0 +1,151 @@
+                ushort ligatureArrayOffset = reader.ReadOffset16();
+
+                var markCoverage = CoverageTable.Load(reader, offset + markCoverageOffset);
+                var ligatureCoverage = CoverageTable.Load(reader, offset + ligatureCoverageOffset);
+                var markArrayTable = new MarkArrayTable(reader, offset + markArrayOffset);
+                var ligatureArrayTable = new LigatureArrayTable(reader, offset + ligatureArrayOffset, markClassCount);
+
+                return new LookupType5Format1SubTable(markCoverage, ligatureCoverage, markArrayTable, ligatureArrayTable);
+            }
+
+            public override bool TryUpdatePosition(
+                IFontMetrics fontMetrics,
+                GPosTable table,
+                GlyphPositioningCollection collection,
+                Tag feature,
+                ushort index,
+                int count)
+            {
+                // Mark-to-Ligature Attachment Positioning.
+                // Implements: https://docs.microsoft.com/en-us/typography/opentype/spec/gpos#lookup-type-5-mark-to-ligature-attachment-positioning-subtable
+                ushort glyphId = collection[index][0];
+                if (glyphId == 0)
+                {
+                    return false;
+                }
+
+                int markIndex = this.markCoverage.CoverageIndexOf(glyphId);
+                if (markIndex == -1)
+                {
+                    return false;
+                }
+
+                // Search backward for a base glyph.
+                int baseGlyphIterator = index;
+                ushort baseGlyphId;
+                while (--baseGlyphIterator >= 0)
+                {
+                    GlyphShapingData data = collection.GetGlyphShapingData(baseGlyphIterator);
+                    baseGlyphId = collection[baseGlyphIterator][0];
+                    if (!fontMetrics.TryGetGlyphClass(baseGlyphId, out GlyphClassDef? glyphClass) && !CodePoint.IsMark(data.CodePoint))
+                    {
+                        break;
+                    }
+
+                    if (glyphClass != GlyphClassDef.MarkGlyph)
+                    {
+                        break;
+                    }
+                }
+
+                if (baseGlyphIterator < 0)
+                {
+                    return false;
+                }
+
+                ushort baseGlyphIndex = (ushort)baseGlyphIterator;
+                baseGlyphId = collection[baseGlyphIndex][0];
+                int ligatureIndex = this.ligatureCoverage.CoverageIndexOf(baseGlyphId);
+                if (ligatureIndex < 0)
+                {
+                    return false;
+                }
+
+                LigatureAttachTable ligatureAttach = this.ligatureArrayTable.LigatureAttachTables[ligatureIndex];
+                ushort markGlyphId = glyphId;
+                ushort ligGlyphId = baseGlyphId;
+
+                // TODO: figure out how to calculate the compIndex, see fontKit.
```

Comment:
not sure howto fix this TODO here. Im not sure what the ligatureID is supposed to be. @JimBobSquarePants maybe you have an idea?

to quote from harfbuzz
```
We must now check whether the ligature ID of the current mark glyph
is identical to the ligature ID of the found ligature.  If yes, we
can directly use the component index.  If not, we attach the mark
glyph to the last component of the ligature.
```

Its `hb-ot-layout-gpos-table.hh` line 1202

Reply:
So, looking at FontKit the ligature Id would be set in a single places. GSUB  `LookupType4SubTable`

The default value (taken from the processor, the `GlyphSubstitutionCollection` in our instance) would be `1`.

![image](https://user-images.githubusercontent.com/385879/136019118-7a7640f6-6220-48d4-8450-fbe8d3e6cf29.png)

They seem to carry a lot of info around with the data right through the process.

![image](https://user-images.githubusercontent.com/385879/136019851-ea835ee7-2525-4089-9b1a-e8873b277083.png)